source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,1896 @@
+  operation. This should correspond to either an `entity`_ name (for
+  `Entity Test Operations`_) or "testRunner" (for `Special Test Operations`_).
+  If the object is an entity, The YAML file SHOULD use an `alias node`_ for its
+  ``id`` field (e.g. ``object: *collection0``).
+
+.. _operation_arguments:
+
+- ``arguments``: Optional document. Map of parameter names and values for the
+  operation. The structure of this document will vary based on the operation.
+  See `Entity Test Operations`_ and `Special Test Operations`_.
+
+  The ``session`` parameter is handled specially (see `commonOptions_session`_).
+
+.. _operation_expectedError:
+
+- ``expectedError``: Optional document. One or more assertions for an expected
+  error raised by the operation. The structure of this document is
+  defined in `expectedError`_.
+
+  This field is mutually exclusive with
+  `expectedResult <operation_expectedResult_>`_ and
+  `saveResultAsEntity <operation_saveResultAsEntity_>`_.
+
+.. _operation_expectedResult:
+
+- ``expectedResult``: Optional mixed type. A value corresponding to the expected
+  result of the operation. This field may be a scalar value, a single document,
+  or an array of documents in the case of a multi-document read.  Test runners
+  MUST follow the rules in `Evaluating Matches`_ when processing this assertion.
+
+  This field is mutually exclusive with
+  `expectedError <operation_expectedError_>`_.
+
+.. _operation_saveResultAsEntity:
+
+- ``saveResultAsEntity``: Optional string. If specified, the actual result
+  returned by the operation (if any) will be saved with this name in the
+  `Entity Map`_.  The test runner MUST raise an error if the name is already in
+  use. If the operation does not return a value (e.g. void method), the test
+  runner MAY choose to store an empty value (e.g. ``null``) or do nothing and
+  leave the entity name undefined.
+
+  This field is mutually exclusive with
+  `expectedError <operation_expectedError_>`_.
+
+  **TODO**: This is primarily used for change streams. Once an operation for
+  iterating a change stream is added, it should link to ``saveResultAsEntity``
+  as this will be the only way to add a change stream object to the entity map.
+
+
+expectedError
+~~~~~~~~~~~~~
+
+One or more assertions for an error/exception, which is expected to be raised by
+an executed operation. At least one key is required in this document.
+
+The structure of this document is as follows:
+
+- ``type``: Optional string or array of strings. One or more classifications of
+  errors, at least one of which should apply to the expected error.
+
+  Valid types are as follows:
+
+  - ``client``: client-generated error (e.g. parameter validation error before
+    a command is sent to the server).
+
+  - ``server``: server-generated error (e.g. error derived from a server
+    response).
+
+- ``errorContains``: Optional string. A substring of the expected error message.
```

Comment:
Did you mean to comment on `errorCodeName`? I don't recall exactly when that was added but it may have been to make the tests more readable (not really an argument since we can just use YAML comments, as we do for fail points to translate the numeric code).

Either way, I don't have a good answer for why tests introduced `errorCodeName` instead of a numeric check with `errorCode`. I'm open to changing this.

Reply:
To follow-up, you _were_ asking about why so many tests check error messages (not code names). Prashant explained this is largely for legacy reasons because early server versions did not have consistent codes. Error message assertions are also helpful for testing client-side errors (when a driver might use a templated error message but not a consistent code -- or any code at all).